frontend/app/profile/edit/page.tsx

```diff
@@ -0,0 +1,279 @@
+import axiosInstance from "@/lib/axios";
+import { PlayerProfile } from "@/types/player-profile";
+import { PlayerProfileRequest } from "@/types/player-profile-request";
+import { AxiosError } from "axios";
+import { useRouter } from "next/navigation";
+import React, { useEffect, useState } from "react";
+import {
+  Card,
+  CardContent,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import { Button } from "@/components/ui/button";
+import ProfilePicture from "@/components/profile/EditProfilePicture";
+import { validateImageFile } from "@/utils/fileValidation";
+
+const API_URL = `${process.env.NEXT_PUBLIC_SPRINGBOOT_API_URL}`;
+const PROFILE_IMAGE_API = `${process.env.NEXT_PUBLIC_PROFILEPICTURE_API_URL}`;
+
+const EditProfile = () => {
+  const { user } = useAuth();
+  const { showErrorToast } = useErrorHandler();
+  const [playerProfileRequest, setPlayerProfileRequest] =
+    useState<PlayerProfileRequest>({
+      id: user?.userId || "",
+      profileUpdates: {} as PlayerProfile,
+      profileImage: new File([], ""),
+    });
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedImage, setSelectedImage] = useState<string | null>(null);
+  const router = useRouter();
+
+  const fetchProfile = async () => {
+    const response = await fetch(`${API_URL}/profile/${user?.userId}`);
+    const data = await response.json();
+
+    if (!data.profilePicturePath) {
+      data.profilePicturePath = PROFILE_IMAGE_API + data.username;
+    }
+
+    setPlayerProfileRequest((prev) => ({
+      ...prev,
+      profileUpdates: data,
+    }));
+
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    fetchProfile();
+  }, []);
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    // Create FormData object
+    const formData = new FormData();
+
+    // Add the id
+    formData.append("id", playerProfileRequest.id);
+
+    // Add the profile updates as a JSON string
+    formData.append(
+      "profileUpdates",
+      new Blob([JSON.stringify(playerProfileRequest.profileUpdates)], {
+        type: "application/json",
+      })
+    );
+
+    // Add the profile image if it exists and is not empty
+    if (playerProfileRequest.profileImage.size > 0) {
+      formData.append("profileImage", playerProfileRequest.profileImage);
+    }
+
+    axiosInstance
+      .put(new URL(`/profile`, API_URL).toString(), formData, {
+        headers: {
+          "Content-Type": "multipart/form-data",
+        },
+      })
+      .then((response) => {
+        if (response.status === 200) {
+          toast({
+            variant: "success",
+            title: "Success",
+            description: "Profile updated.",
+          });
+          router.push("/profile/" + user?.userId);
+        }
+      })
+      .catch((error: AxiosError) => {
+        let title = "Error";
+        let message = error.message;
+
+        if (error.response) {
+          switch (error.response.status) {
```

Comment:
should refactor and separate this message into alert dialog

Reply:
Will be refactored in the next sprint